src/kubernetes.cc

```diff
@@ -823,12 +822,26 @@ void KubernetesReader::WatchMaster(
       if (verbose) {
         LOG(INFO) << "Resetting and running service";
       }
+      if (watch_service->stopped()) {
```

Comment:
It seems like there could be a race condition between here and :835, is this true, and if so, how could we address it?

Reply:
Why would there be a race condition? The delay timer will kick off only after this test, and then code will not loop around here until it gets past the `join()` in line 849. The `join()` will block until the watcher exits, which it will after the delay timer callback fires. Am I missing something here?